pkg/operator/controller/credentials-secret/credentials_secret.go

```diff
@@ -124,11 +128,27 @@ func desiredCredentialsSecret(sourceSecret *corev1.Secret, destName types.Namesp
 		case operatorv1alpha1.ProviderTypeAWS:
 			secret.Data = sourceSecret.Data
 		}
-	} else {
-		secret.Data = sourceSecret.Data
+		return secret, nil
 	}
 
-	return secret
+	if extDNS.Spec.Provider.Type == operatorv1alpha1.ProviderTypeAWS {
+		secret.Data = make(map[string][]byte)
```

Comment:
Not needed, `Data` field is already initialized at the beginning of the function.